### DISCUSSION TITLE
Consistency check

### DISCUSSION THREAD

**Discussion by mitsu-ko** [TPC-Council/HammerDB#686] [Q&A] [answered]
Hi,

I use consistency check feature in HammerDB 4.10 on PostgreSQL 15.6. Before benchmark, 'checkschema' is done without error. After running TPC-C benchmark, I run 'checkschame', consistency check fail is outputted under following. 

Vuser 1 created - WAIT IDLE
Vuser 1:RUNNING
Vuser 1:Checking tpcc TPROC-C schema
Vuser 1:Check database
Vuser 1:Check tables and indices
Vuser 1:Check procedures
Vuser 1:Check consistency 1
Vuser 1:Check consistency 2
Error in Virtual User 1: TPROC-C Schema check failed tpcc schema consistency check 2 failed
Vuser 1:FINISHED FAILED

Does it means HammerDB bug or PostgreSQL bug? And I'd like to know more detail information why check is failed. It's important problem. But error message is too simple.

Regards,

**Comment by sm-shaw** [comment]
I can reproduce, and this is a HammerDB bug. I will create an issue from the discussion for the fix. 
The bug/fix will not have any impact on test performance.

**Comment by sm-shaw** [comment, marked as answer]
Issue https://github.com/TPC-Council/HammerDB/issues/687 has been raised with further explanation.

**Reply by sm-shaw** [reply]
PR https://github.com/TPC-Council/HammerDB/pull/688 fixes the issue and the consistency checks run successfully afterwards.

**Comment by mitsu-ko** [comment]
Hi sm-shaw-san,

Thank you very much for your fast work. I think consistency check feature is very interesting. I appreciate for your work.

Regards,

**Reply by sm-shaw** [reply]
No problem, and thank-you for raising the issue. I am not entirely sure how this bug got through but that is what the consistency checks are for to identify issues in both the HammerDB and database code.  For example, when testing distributed databases and replication it will be a good check to ensure that all parts of the system are working.
